dot/state/storage_test.go

```diff
@@ -227,7 +228,12 @@ func TestGetStorageChildAndGetStorageFromChild(t *testing.T) {
 		"0",
 	))
 
-	testChildTrie := trie.NewEmptyTrie()
+	key := []byte{1, 2}
+	value := []byte{3, 4}
+	const dirty = true
+	const generation = 0
+	testChildTrie := trie.NewTrie(node.NewLeaf(key, value, dirty, generation))
```

Comment:
nit a bit simpler, since now most node fields are exported (for testing really) so `NewLeaf` is to be deprecated

```suggestion
	childTrieRoot := &node.Leaf{
		Key: []byte{1, 2},
		Value: []byte{3, 4}
		Dirty: true,
		Generation: 0,
	}
	testChildTrie := trie.NewTrie(childTrieRoot)
```

Reply:
trie.NewTrie is now inside a loop, so these variables are needed there.

Reply:
They aren't used anywhere, I think you are mistaking this with another test? :thinking:

Reply:
I prefer using constructors whenever we can even if the fields are exported.